src/usage/search.md

```diff
@@ -3,38 +3,81 @@
 The SQL for searching is very simple. Here is an example of searching the $5$ nearest embedding in table `items`:
 
 ```sql
+SET vectors.hnsw_ef_search = 64;
 SELECT * FROM items ORDER BY embedding <-> '[3,2,1]' LIMIT 5;
 ```
 
-::: info
+The vector index will search for `64` nearest rows, and `5` nearest rows is gotten since there is a `LIMIT` clause.
 
-If `vectors.k` is set to `64`, but your SQL returned less than `64` rows, for example, only `32` rows. There is some possible reasons:
+## Search modes
 
-* Less than `64` rows should be returned. It's expected.
-* The vector index returned `64` rows, but `32` of which are deleted before but the index do not know since PostgreSQL vacuum is lazy.
-* The vector index returned `64` rows, but `32` of which are invisible to the transaction so PostgreSQL decided to hide these rows for you.
-* The vector index returned `64` rows, but `32` of which are satisfying the condition `id % 2 = 0` in `WHERE` clause.
+There are two search modes: `basic` and `vbase`.
 
-There are four ways to solve the problem:
+### `basic`
 
-* Set `vectors.k` larger. If you estimate that 20% of rows will satisfy the condition in `WHERE`, just set `vectors.k` to be 5 times than before.
-* Set `vectors.enable_vector_index` to `off`. If you estimate that 0.0001% of rows will satisfy the condition in `WHERE`, just do not use vector index. No algorithms will be faster than brute force by PostgreSQL.
-* Set `vectors.enable_prefilter` to `on`. If you cannot estimate how many rows will satisfy the condition in `WHERE`, leave the job for the index. The index will check if the returned row can be accepted by PostgreSQL. However, it will make queries slower so the default value for this option is `off`.
-* Set `vectors.enable_vbase` to `on`. It will use vbase optimization, so that the index will pull rows as many as you need. It only works for HNSW algorithm.
+`basic` is the default search mode. In this mode, vector indexes behave like a vector search library. It works well if all of your queries is like this:
 
-:::
+```sql
+SELECT * FROM items ORDER BY embedding <-> '[3,2,1]' LIMIT 5;
+```
+
+It's recommended if your do **not** take advantages of
+
+* database transaction
+* deletions without `VACUUM`
+* WHERE clauses and very complex SQL statments
+
+### `vbase`
+
+`vbase` is another search mode. In this mode, vector indexes behave like a database index.
```

Comment:
Could we add the link to the paper here?

Reply:
And could we explain more about `vector indexes behave like a database index.`

Reply:
PTAL